### DISCUSSION TITLE
How to write a schedule that launches a single run that fills a range of asset partitions?

### DISCUSSION THREAD

**Discussion by sryza** [dagster-io/dagster#14622] [Q&A] [answered]
Every night, I want to launch a single run that materializes a range of asset partitions.

**Comment by sryza** [comment, marked as answer]
You can accomplish this with a `RunRequest` that supplies the "dagster/asset_partition_range_start" and "dagster/asset_partition_range_end" tags:

```python
@schedule(job=..., cron_schedule="@daily")
def my_schedule():
    ...
    return RunRequest(
        tags={
            "dagster/asset_partition_range_start": start_partition_key,
            "dagster/asset_partition_range_end": end_partition_key,
        }
    )
```

**Reply by geoHeil** [reply]
But this is not applicable for the DBT API? at least not by default?

**Reply by sryza** [reply]
@geoHeil that's correct: dbt assets are not partitioned by default, and after making them partitioned, you need to define how the partitions get translated to filters in the dbt statements.

**Reply by PatriciaAyuso** [reply]
@sryza @geoHeil 
Here is a conversation about this https://github.com/dagster-io/dagster/issues/7683#issuecomment-1169473649

**Reply by ytoast** [reply]
Would this work for static partitions as well?

**Reply by sryza** [reply]
@ytoast it should

**Comment by garethbrickman** [comment]
To write a sensor or schedule that requests a run for every partition on every tick, see https://github.com/dagster-io/dagster/discussions/15532

**Comment by AdrianTheopold** [comment]
I have an asset with a dynamic partition that is using ` context.partition_keys`. 
How can I write a schedule/sensor (or a general python script for that matter) that runs the asset creation for a large number of partitions, say 2000 in my case.

I am able to run backfills through the UI which is very inconvenient for a large selection of partitions. But other than that I don't see an option for creating such a backfill via a sensor or script.

`$ dagster job backfill -j do_stuff_partitioned --partitions`
does not work for such a large number of partitions.
